json.go

```diff
@@ -0,0 +1,33 @@
+package elation
+
+import "encoding/json"
+
+// NonNullJSONArray ensures that nil slices are marshalled to "[]" instead of "null". It also ensures that empty JSON arrays
+// are marshalled to nil slices.
+type NonNullJSONArray[T any] []T
+
+func (nnja NonNullJSONArray[T]) MarshalJSON() ([]byte, error) {
+	if nnja == nil {
+		nnja = make([]T, 0)
+	}
+
+	return json.Marshal([]T(nnja))
+}
+
+func (nnja *NonNullJSONArray[T]) UnmarshalJSON(data []byte) error {
+	type alias NonNullJSONArray[T]
+	var a alias
```

Comment:
Is this type alias necessary?

```suggestion
	var a NonNullJSONArray[T]
```

Reply:
Yes, the alias is needed to prevent infinite recursion on UnmarshalJSON.

Reply:
I'll add test coverage so you can check for yourself.